src/file-events/cpp/linux_fsnotifier.cpp

```diff
@@ -155,21 +166,29 @@ void Server::processCommandsOnThread() {
 
 void Server::handleEvent(JNIEnv* env, const inotify_event* event) {
     uint32_t mask = event->mask;
-    log_fine(env, "Event mask: 0x%x for %s (wd = %d, cookie = 0x%x)", mask, event->name, event->wd, event->cookie);
+    const char* eventName = (event->len == 0)
+        ? ""
+        : event->name;
+    log_fine(env, "Event mask: 0x%x for %s (wd = %d, cookie = 0x%x, len = %d)", mask, eventName, event->wd, event->cookie, event->len);
     if (IS_ANY_SET(mask, IN_UNMOUNT)) {
         return;
     }
-    // TODO Do we need error handling here?
+
     u16string path = watchRoots[event->wd];
+    if (path.empty()) {
```

Comment:
So if an element is not part of the map, then an empty String is returned? Because it uses the empty constructor of `u16string`?

Reply:
That's my interpretation, too, and it matches what I've seen. It's weird.